common/tree/script.go

```diff
@@ -414,30 +406,39 @@ func (d *CSVMultisigClosure) Script() ([]byte, error) {
 }
 
 func (d *CSVMultisigClosure) Decode(script []byte) (bool, error) {
-	if len(script) == 0 {
+	tokenizer := txscript.MakeScriptTokenizer(0, script)
+
+	if len(tokenizer.Script()) == 0 {
 		return false, fmt.Errorf("empty script")
 	}
 
-	csvIndex := bytes.Index(
-		script, []byte{txscript.OP_CHECKSEQUENCEVERIFY, txscript.OP_DROP},
-	)
-	if csvIndex == -1 || csvIndex == 0 {
+	if tokenizer.Err() != nil {
 		return false, nil
 	}
 
-	sequence := script[:csvIndex]
-	if len(sequence) > 5 {
-		return false, fmt.Errorf("invalid sequence length")
+	_ = tokenizer.Next()
```

Comment:
Why do we "skip" a token here ? Maybe worth to put a small comment explaining this